### DISCUSSION TITLE
Flux's Image Policy Roadmap?

### DISCUSSION THREAD

**Discussion by francoisfaubert** [fluxcd/flux2#1728] [General] [answered]
We have been using Flux to manage our clusters for a couple of months now. We use image policies to automate updates of services accross all of them and things work great on the K8s side.

One thing that has not been working as great is the cleanup lifecycle of our containers on AWS ECR. Since we tag using the proposed `<branch>-<sha1>-<timestamp>` format, our containers tags cannot be grouped as older versions of the same thing by AWS's lifecycle policies (as far as I know). This prevents us from expiring our older versions as we go.

I would like to know is whether there is a long term plan for having Flux image policies that could handle version changes simply by using `<branch>`, say. I know of the rate limits on container hubs, but maybe you have a plan around these already.

Thanks!

**Comment by stefanprodan** [comment, marked as answer]
Flux can't work with mutable tags e.g. `latest`, `main`, etc. Each tag must be immutable and unique for GitOps to make sense.
